disnake/ui/modal.py

```diff
@@ -47,71 +49,51 @@ class Modal:
 
     Parameters
     ----------
-    title: str
+    title: :class:`str`
         The title of the modal.
-    custom_id: str
+    custom_id: :class:`str`
         The custom ID of the modal.
-    components: List[:class:`~.ui.InputText`]
-        The components to display in the modal. Maximum of 5.
+    components: |components_type|
+        The components to display in the modal. Up to 5 action rows.
+    timeout: :class:`float`
+        Timeout from the moment this modal is sent.
+        Defaults to 600 seconds. Modals without timeouts are not supported,
+        since there's no event for closing modals.
```

Comment:
```suggestion
    timeout: :class:`float`
        The time to wait until the modal is removed from cache, if no interaction is made.
        Modals without timeouts are not supported, since there's no event for when a modal is closed.
        Defaults to 600 seconds.
```
We can discuss this change if you want.